packages/zoe/src/contracts/operaConcertTicket.js

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable no-use-before-define */
+import harden from '@agoric/harden';
+import produceIssuer from '@agoric/ertp';
+import { makeZoeHelpers } from './helpers/zoeHelpers';
+
+/*
+  Roles in the arrangement:
+  - Contract creator: describes the contract with:
+    - number of seats, show, date/time of start
+    - expected (ERTP) amount per ticket (we assume all tickets cost the same)
+  - Smart Contract: 
+    - mints the tickets
+    - provides the seats
+  - Auditorium (unique contract seat, usually taken by the contract creator): the person hosting 
+  the Opera show, selling the tickets and getting the payment back
+  - Ticket buyers (contract seat created on demand): 
+    - can see the available opera show seats
+    - can consult the terms
+    - can redeem the zoe invite with the proper payment to get the ticket back
+*/
+
+export const makeContract = harden(async zoe => {
+  // Create the internal ticket mint
+  const { issuer, mint, amountMath } = produceIssuer('Opera tickets', 'set');
+  await zoe.addNewIssuer(issuer, 'Ticket');
+
+  // create Zoe helpers after zoe.addNewIssuer because of https://github.com/Agoric/agoric-sdk/issues/802
+  const { rejectOffer } = makeZoeHelpers(zoe);
+
+  const {
+    terms: { show, start, count },
+  } = zoe.getInstanceRecord();
+
+  const inviteHandleByTicketNumber = new Map();
+
+  function completeAmountKeywordRecord(amountKeywordRecord) {
```

Comment:
I think we can take this out, since it is recreating the amountKeywordRecord that exists as `offerRecord.amounts`.